src/App.tsx

```diff
@@ -1,26 +1,148 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Errors } from './types/Errors';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Empty);
+  const [status, setStatus] = useState<string>(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [isDeletingComleted, setIsDeletingCompleted] = useState(false);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(data => setTodos(data))
+      .catch(() => {
+        setErrorMessage(Errors.UnableToLoad);
+      });
+  }, []);
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = todos.filter(todo => {
+    if (
+      status === Status.All ||
+      (status === Status.Completed && todo.completed) ||
+      (status === Status.Active && !todo.completed)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
the same( `||` return boolean in this case)

```suggestion
      status === Status.All ||
      (status === Status.Completed && todo.completed) ||
      (status === Status.Active && !todo.completed)
```